        protected void signalCloseDone()
        {
            closeFuture.get().set(null);

            // We can now close the socket
            try
            {
                socket.close();
            }
            catch (IOException e)
            {
                // Erroring out while closing shouldn't happen but is not really a big deal, so just log
                // it at DEBUG and ignore otherwise.
                logger.debug("Unexpected error while closing streaming connection", e);
            }
        }

